### DISCUSSION TITLE
How to calculate natural Logarithm in Script?

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#11501] [Q&A] [answered]
Hello everyone,
i was wondering how i could calcualte an natural logarigthm in a Script.
I didnt find anything about that in the Documentation.

Thanks in advance.

**Comment by sfromis** [comment, marked as answer]
AFAICT, not currently available. A couple of places, other Tasmota code use a compact but limited Taylor series logarithm function, but that can be quite imprecise outside of specific use cases. Due to space constraints, using "full" math functions have been avoided.
